### DISCUSSION TITLE
Welcome to vscode-xp Discussions!

### DISCUSSION THREAD

**Discussion by aw350m33d** [Security-Experts-Community/vscode-xp#1] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by Ideas4Life** [comment]
Добрый вечер, начал тестить расширение для VS Code, но столкнулся с проблемами при настройке. Можете подсказать, что не так делаю:
1. Сказано, что надо настроить путь к утилите kbtools. Какой параметр в настройках расширения за это отвечает? Догадки есть, но как проверить не знаю.
2. Также сказано, что "нужно нажать кнопку Экспорт kb-файла". Где эта кнопка? Я вижу кнопку, чтобы открыть уже распакованную базу знаний (не в формате kb)
3. Откуда берётся файл "tables_contract.yaml"? При создании с нуля написано, что он автоматом создаётся - пока тоже непонятно как создать с нуля базу, так как кнопки нет. А если берём готовую базу, то расширение само формирует этот файл или как? Просто при ручном использовании kbtools не припомню этого файла среди получаемых файлов.
![image](https://user-images.githubusercontent.com/47158221/220258285-364a2c98-a31f-44a2-a914-21f1b53a3ba5.png)
![image](https://user-images.githubusercontent.com/47158221/220258307-162c5cd8-3cff-471f-933b-960b1483bf81.png)

**Reply by aw350m33d** [reply]
Привет! Спасибо за вопрос, мы обязательно сделаем более детальное описание настройки и использования плагина.

## 1) Путь к kbtools.exe
Путь **к папке** с `kbtools.exe` указывается в параметре `Knowledge Packager Directory Path`:
![image](https://user-images.githubusercontent.com/61383585/220392920-1f2425d8-e8dd-4796-a383-c0b566da5421.png)

## 2) Распаковка пакета
Для этого нужно произвести небольшую подготовку. В будущем реализуем фичу по автоматическому выполнению такой подготовки.

Если кратко, то чтобы появилась описанная "кнопка", нужно нажать правой кнопкой мыши по папке packages, которая должна находиться внутри открытой базы знаний.

Поэтому тебе нужно:
1. Создать папку packages
![image](https://user-images.githubusercontent.com/61383585/220396105-5745f443-74cd-4c13-8e86-2e7b990e33b7.png)
![image](https://user-images.githubusercontent.com/61383585/220396191-281d8a7a-b1d4-444f-97f5-8bd0d7fb7761.png)

2. Нажать правой кнопкой на папку packages и выбрать пункт `Извлечь пакет(ы) из kb-файла`
![image](https://user-images.githubusercontent.com/61383585/220396535-ee8a1a4d-411e-4f10-b0fa-3f823a25a01c.png)

3. В появившемся диалоге выбора файла найти *.kb-файл

5. После выбора файла сразу же начнётся процесс распаковки. По окончании появится сообщение об успешности:
![image](https://user-images.githubusercontent.com/61383585/220397164-0ba2533b-ff09-4dba-961d-7071c7be6e91.png)

6. Если всё прошло успешно, то внутри папки packages появится распакованный пакет.
![image](https://user-images.githubusercontent.com/61383585/220397416-f775c9b9-aef6-4050-9f1b-0e775ef95317.png)


## 3) Файлик tables_contract.yaml
Пока этого файлика нет в открытом доступе. Мы активно работаем над тем, чтобы подготовить к публикации и утилиты SDK и все сопутствующие файлы. А пока можно использовать вот этот:
[tables_contract.zip](https://github.com/Security-Experts-Community/vscode-xp/files/10795488/tables_contract.zip)

**Reply by Ideas4Life** [reply]
Добрый вечер!
Спасибо за ответ. Частично получилось
К сожалению, при импорте всей базы или пакеты экспертизы "Базовый пакет" отображается ошибка. При этом пользовательский пакет экспертизы или иной стандартный пакет экспертизы (например, "Active Directory") успешно импортируются. В чем может быть дело?
![image](https://user-images.githubusercontent.com/47158221/220715069-e32fb9db-e648-48d1-873a-86deaee5660c.png)

**Reply by aw350m33d** [reply]
Добрый вечер!  
Данная ошибка относится к утилите kbtools, которая является внешней утилитой для нашего плагина и поддерживается командой разработки MP SIEM. Мы выйдем с ними на связь и выясним в чём может быть причина. А пока я заведу issue со ссылкой на Ваше сообщение. Отслеживать статус и вести дальнейшее обсуждение этой конкретной проблемы предлагю там.

**Reply by aw350m33d** [reply]
Вот созданная issue:
https://github.com/Security-Experts-Community/vscode-xp/issues/8
